test/support/otlp_endpoint_test.rb

```diff
@@ -0,0 +1,228 @@
+require './lib/solarwinds_apm/support/service_key_checker'
+require './lib/solarwinds_apm/support/utils'
+
+# BUNDLE_GEMFILE=gemfiles/unit.gemfile bundle exec ruby -I test test/support/otlp_endpoint_test.rb
+describe 'OTLP Endpoint Test' do
+  before do
+    @original_env = ENV.to_h.dup
+    ENV.clear
+  end
+
+  after do
+    ENV.replace(@original_env)
+  end
+
+  describe 'config_token' do
+    def token_setup
+      @endpoint = SolarWindsAPM::OTLPEndPoint.new
+      @endpoint.resolve_service_key
+      %w[TRACES METRICS LOGS].each { |data_type| @endpoint.config_token(data_type) }
+    end
+
+    def assert_entity_endpoint_nil
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_TRACES_ENDPOINT', nil))
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_METRICS_ENDPOINT', nil))
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_LOGS_ENDPOINT', nil))
+    end
+
+    def assert_headers_nil
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_METRICS_HEADERS', nil))
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_TRACES_HEADERS', nil))
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_LOGS_HEADERS', nil))
+    end
+
+    it 'wrong formatted SW_APM_SERVICE_KEY OTEL PROTO' do
+      ENV['SW_APM_SERVICE_KEY'] = nil
+      ENV['OTEL_EXPORTER_OTLP_ENDPOINT'] = 'https://otel.collector.na-01.cloud.solarwinds.com:443'
+
+      token_setup
+
+      assert_nil(@endpoint.instance_variable_get(:@token))
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_HEADERS', nil))
+
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_ENDPOINT', nil)).must_equal 'https://otel.collector.na-01.cloud.solarwinds.com:443'
+
+      assert_entity_endpoint_nil
+      assert_headers_nil
+    end
+
+    it 'correct formatted SW_APM_SERVICE_KEY, auth token set for OTEL PROTO' do
+      ENV['SW_APM_SERVICE_KEY'] = '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234:my_service'
+      ENV['OTEL_EXPORTER_OTLP_ENDPOINT'] = 'https://otel.collector.na-01.cloud.solarwinds.com:443'
+
+      token_setup
+
+      _(@endpoint.instance_variable_get(:@token)).must_equal '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234'
+
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_HEADERS', nil)).must_equal 'authorization=Bearer 0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234'
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_ENDPOINT', nil)).must_equal 'https://otel.collector.na-01.cloud.solarwinds.com:443'
+
+      assert_entity_endpoint_nil
+      assert_headers_nil
+    end
+
+    it 'wrong formatted SW_APM_SERVICE_KEY APM PROTO' do
+      ENV['SW_APM_SERVICE_KEY'] = nil
+      ENV['SW_APM_COLLECTOR'] = 'apm.collector.na-01.cloud.solarwinds.com:443'
+
+      token_setup
+
+      assert_nil(@endpoint.instance_variable_get(:@token))
+
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_ENDPOINT', nil))
+      assert_entity_endpoint_nil
+      assert_headers_nil
+    end
+
+    it 'correct formatted SW_APM_SERVICE_KEY, auth token set for OTEL PROTO resolved from APM COLLECTOR config' do
```

Comment:
```suggestion
    it 'correct formatted SW_APM_SERVICE_KEY APM PROTO, no auth token set due to no OTEL PROTO' do
```

i find this test just really confusing and have to keep remembering that we're not configuring the OTLP exporter thus header is not set.